pandas/tests/indexing/multiindex/test_getitem.py

```diff
@@ -279,3 +280,32 @@ def test_loc_empty_multiindex():
     result = df
     expected = DataFrame([1, 2, 3, 4], index=index, columns=["value"])
     tm.assert_frame_equal(result, expected)
+
+
+@pytest.mark.parametrize("nan", [np.nan, pd.NA, None])
+def test_loc_nan_multiindex(nan):
+    # GH#29751
+    # tests to check loc on a multiindex containing nan values
+    arr = [
+        [11, nan, 13],
+        [21, nan, 23],
+        [31, nan, 33],
+        [41, nan, 43],
+        [51, nan, 53],
+    ]
+    cols = ["a", "b", "c"]
+    df = DataFrame(arr, columns=cols, dtype="int").set_index(["a", "b"])
```

Comment:
use dtype='int64' to avoid the 32-bit failures

Reply:
Changed dtype to int64 in the recent commit